modules/caddyhttp/matchers.go

```diff
@@ -470,13 +470,28 @@ func (m *MatchHeader) UnmarshalCaddyfile(d *caddyfile.Dispenser) error {
 	}
 	for d.Next() {
 		var field, val string
-		if !d.Args(&field, &val) {
-			return d.Errf("malformed header matcher: expected both field and value")
+		if !d.Args(&field) {
+			return d.Errf("malformed header matcher: expected field")
 		}
 
-		// If multiple header matchers with the same header field are defined,
-		// we want to add the existing to the list of headers (will be OR'ed)
-		http.Header(*m).Add(field, val)
+		if strings.HasPrefix(field, "!") {
+			field = strings.TrimPrefix(field, "!")
```

Comment:
```suggestion
			field = field[1:]
```

This way we don't have to repeat the magic `!` symbol.